### PR TITLE
TE-1904 Persist search value with a cookie

### PR DESCRIPTION
**Pull request by red-runner**
[Related YouTrack issue](https://youtrack.lodgify.net/issue/TE-1904)

### What are the 3 things this PR does?

- Removes unnecessary `defaultValue` from `Dropdown`.
- Adds initial values for `SearchBar` state inputs.
- Fixes `HomepageHero` and `PropertySearchBar` guest input PropTypes.

### CODE REVIEW

File: src/components/general-widgets/HomepageHero/component.js [hunk 1]
```diff
@@ -236,7 +236,11 @@ Component.propTypes = {
    */
   searchBarGetIsDayBlocked: PropTypes.func,
   /** The initial value for the guests input of the search bar. */
-  searchBarGuestsInputInitialValue: PropTypes.string,
+  searchBarGuestsInputInitialValue: PropTypes.oneOfType([
+    PropTypes.bool,
```
Comment (Jonnygwi) [inline src/components/general-widgets/HomepageHero/component.js]: When is this value a `boolean`?

File: src/components/general-widgets/SearchBar/component.js [hunk 2]
```diff
@@ -20,7 +20,11 @@ import { getSearchBarModal } from './utils/getSearchBarModal';
  */
 // eslint-disable-next-line jsdoc/require-jsdoc
 export class Component extends PureComponent {
-  state = {};
+  state = {
+    dates: { endDate: null, startDate: null },
+    guests: null,
+    location: null,
+  };
```
Comment (marcdavi-es) [inline src/components/general-widgets/SearchBar/component.js]: Doesn't this need to be populated with the initial values if they are available? Or does it get those on the first update?

File: src/components/general-widgets/HomepageHero/component.js (hunk 1 above)
Comment (red-runner) [reply src/components/general-widgets/HomepageHero/component.js]: @Jonnygwi I am slightly uncertain for this as well. I've checked elsewhere, like in the `Footer` component and I found this: 
```
  /** The current value of the language dropdown. */
  languageValue: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.number,
    PropTypes.string,
  ]),
```
What do you think?

**Pull request merged**